### DISCUSSION TITLE
What is the intent of storing the git hash in the SSI field?

### DISCUSSION THREAD

**Discussion by mhardin18** [IBM/dbb-zappbuild#469] [Q&A] [answered]
I want to understand the intent behind storing the abbreviated git hash in the SSI parm for the cobol link?  

This produces output into the shiplist file if you are using the dbb project pipeline scripts. This in turn has a side effect in UCD. UCD interprets that SSI parameter as a Last Modified date and will then interpret the abbreviated git has string as some arbitrary date such as 05/30/2071.

So why are we storing the git hash in the SSI parm?

This behavior is controlled by the cobol_storeSSI property and applies to other languages as well.

**Comment by dennis-behm** [comment, marked as answer]
Hi Matthew,

the idea with storing short git hash in the SSI field is to be able to quickly identify the state of the application repository from where the module got built from. 

We are aware that this conflicts with how UCD is interpreting the SSI. UCD is using this field for a different purpose (capturing it as a timestamp).

We have users who modified the data that is written into the SSI field. Converting the build timestamp and use it as the SSI input will provide you basically the build timestamp.

Alternatively, you can turn this functionality off using the provided properties :

https://github.com/IBM/dbb-zappbuild/blob/dd59b05112791ef1ea5fc710d821cb214b90142d/samples/application-conf/Cobol.properties#L60-L64
